scripts/changelog

```diff
@@ -0,0 +1,326 @@
+#!/usr/bin/env python
+
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A simple changelog generator
+
+This tool takes list of release versions to generate `CHANGELOG.md`.
+The changelog will include all merged PRs w/o `[bot]` postfix,
+and issues that include the labels `bug`, `feature request`, `SQL`, `performance` and `shuffle`,
+minus any issues with the labels `wontfix`, `invalid` or `duplicate`.
+
+For each project there should be an issue subsection for,
+Features: all issues with label `feature request` + `SQL`
+Performance: all issues with label `performance` + `shuffle`
+Bugs fixed: all issues with label `bug`
+
+To deduplicate section, the priority should be `Bugs fixed > Performance > Features`
+
+NOTE: This is a repo-specific script, so you may not use it in other places.
+Release version from arguments will be used to map project name and branch name.
+Mapping Pattern:
+release -> project: Release {release}, branch: branch-{release}
+e.g.
+    0.1 -> project: Release 0.1, branch: branch-0.1
+    0.2 -> project: Release 0.2, branch: branch-0.2
+
+Dependencies:
+	- requests
+
+Usage:
+    cd spark-rapids/
+
+	# generate changelog for release 0.1,0.2
+	scripts/changelog --token=<GITHUB_PERSONAL_ACCESS_TOKEN> --releases=0.1,0.2
+
+	# generate changelog for release 0.1,0.2 to /tmp/CHANGELOG.md
+    GITHUB_TOKEN=XXX scripts/changelog --releases=0.1,0.2 --path=/tmp/CHANGELOG.md
+"""
+import os
+import sys
+from argparse import ArgumentParser
+from collections import OrderedDict
+from datetime import date
+
+import requests
+
+parser = ArgumentParser(description="Changelog Generator")
+parser.add_argument("--releases", help="list of release versions, separated by comma",
+                    default="0.1,0.2")
+parser.add_argument("--token", help="github token, will use GITHUB_TOKEN if empty", default='')
+parser.add_argument("--path", help="path for generated changelog file", default='./CHANGELOG.md')
+args = parser.parse_args()
+
+GITHUB_TOKEN = args.token if args.token else os.environ.get('GITHUB_TOKEN')
+assert GITHUB_TOKEN, 'env GITHUB_TOKEN should not be empty'
```

Comment:
All of the code related to building and parsing arguments needs to not be at the top-level of the module, otherwise there's little point in the `__name__ == '__main__'` check at the end because importing this as a module will simply fail in many contexts due to argument parse fail.

Ideally `main()` should take an args parameter and this code should be within the `__name__ == '__main__'` block to build the arguments to pass to `main()`.

Reply:
yeah, this code was not supposed to be used as a library. But I agree to make it less janky :P 

new commit updated